homeassistant/components/devolo_home_control/__init__.py

```diff
@@ -47,11 +60,20 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         uuid = await hass.async_add_executor_job(mydevolo.uuid)
         hass.config_entries.async_update_entry(entry, unique_id=uuid)
 
+    def shutdown(event: Event) -> None:
+        for gateway in entry.runtime_data.gateways:
+            gateway.websocket_disconnect(
+                f"websocket disconnect requested by {EVENT_HOMEASSISTANT_STOP}"
+            )
+
+    # Listen when EVENT_HOMEASSISTANT_STOP is fired
+    listener = hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, shutdown)
```

Comment:
```suggestion
    entry.async_on_unload(hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, shutdown))
```
Use `entry.async_on_unload` to now have to store the callback listener for later.